### ISSUE TITLE
Interactive matplotlib plot first gives an error and then grows infinitely

### ISSUE THREAD

**Issue by mgunyho** [marimo-team/marimo#978]
### Describe the bug

When I run `mo.mpl.interactive(ax)`, I first get an asyncio error:

```
Traceback (most recent call last):
  File "~/.virtualenvs/marimo-test/lib/python3.8/site-packages/marimo/_runtime/cell_runner.py", line 228, in run
    return_value = execute_cell(cell, self.glbls)
  File "~/.virtualenvs/marimo-test/lib/python3.8/site-packages/marimo/_ast/cell.py", line 394, in execute_cell
    return eval(cell.last_expr, glbls)
  File "/tmp/marimo_1632908/__marimo__cell_MJUe__output.py", line 3, in <module>
  File "~/.virtualenvs/marimo-test/lib/python3.8/site-packages/marimo/_plugins/stateless/mpl/_mpl.py", line 267, in interactive
    application = get_or_create_application()
  File "~/.virtualenvs/marimo-test/lib/python3.8/site-packages/marimo/_plugins/stateless/mpl/_mpl.py", line 215, in get_or_create_application
    asyncio.run(asyncio.sleep(0.02))
  File "/usr/lib/python3.8/asyncio/runners.py", line 33, in run
    raise RuntimeError(
RuntimeError: asyncio.run() cannot be called from a running event loop
```

The terminal from which I launched marimo also displays the following:

```
~/.virtualenvs/marimo-test/lib/python3.8/site-packages/jedi/inference/compiled/subprocess/__init__.py:220: RuntimeWarning: coroutine 'sleep' was never awaited
  inference_state_id = self._inference_state_deletion_queue.pop()
RuntimeWarning: Enable tracemalloc to get the object allocation traceback
/usr/lib/python3.8/sre_parse.py:433: RuntimeWarning: coroutine 'sleep' was never awaited
  return list(dict.fromkeys(items))
RuntimeWarning: Enable tracemalloc to get the object allocation traceback
```

However, if I re-run the cell after this, the matplotlib figure appears, but starts growing infinitely. Here's a video of it happening:

[https://github.com/marimo-team/marimo/assets/20118130/6e0477d0-96c1-4545-a3ae-8c2d825c40f1](https://github.com/marimo-team/marimo/assets/20118130/6e0477d0-96c1-4545-a3ae-8c2d825c40f1)

This is in a fresh virtualenv where I have only done `pip install marimo matplotlib tornado`. Here's the full output of `pip list`:
Click to open

```
Package             Version    
------------------- -----------
anyio               4.3.0      
black               24.3.0     
click               8.1.7      
contourpy           1.1.1      
cycler              0.12.1     
docutils            0.20.1     
exceptiongroup      1.2.0      
fonttools           4.50.0     
h11                 0.14.0     
idna                3.6        
importlib-metadata  7.1.0      
importlib-resources 6.4.0      
jedi                0.19.1     
kiwisolver          1.4.5      
marimo              0.3.3      
Markdown            3.6        
matplotlib          3.7.5      
mypy-extensions     1.0.0      
numpy               1.24.4     
packaging           24.0       
parso               0.8.3      
pathspec            0.12.1     
pillow              10.2.0     
pip                 20.0.2     
pkg-resources       0.0.0      
platformdirs        4.2.0      
pygments            2.17.2     
pymdown-extensions  10.7.1     
pyparsing           3.1.2      
python-dateutil     2.9.0.post0
PyYAML              6.0.1      
setuptools          44.0.0     
six                 1.16.0     
sniffio             1.3.1      
starlette           0.37.2     
tomli               2.0.1      
tomlkit             0.12.4     
tornado             6.4        
typing-extensions   4.10.0     
uvicorn             0.29.0     
websockets          12.0       
zipp                3.18.1     
```

This is on Python 3.8, which I know is a bit old, but that is what I have available (Ubuntu 20.04). I'm using Firefox 123.0.1, but this also happens on Google Chrome 123.0.6312.58.

### Environment

```json
{
  "marimo": "0.3.3",
  "OS": "Linux",
  "OS Version": "5.15.0-100-generic",
  "Processor": "x86_64",
  "Python Version": "3.8.10",
  "Binaries": {
    "Chrome": "123.0.6312.58",
    "Node": "--"
  },
  "Requirements": {
    "black": "24.3.0",
    "click": "8.1.7",
    "jedi": "0.19.1",
    "pygments": "2.17.2",
    "pymdown-extensions": "10.7.1",
    "starlette": "0.37.2",
    "tomlkit": "0.12.4",
    "uvicorn": "0.29.0"
  }
}
```

### Code to reproduce

```python
import marimo

__generated_with = "0.3.3"
app = marimo.App()


@app.cell
def __():
    import marimo as mo
    import matplotlib.pyplot as plt
    return mo, plt


@app.cell
def __(mo, plt):
    ax = plt.figure().gca()
    ax.plot([1, 2, 3], [4, 5, 4])
    mo.mpl.interactive(ax)
    return ax,


if __name__ == "__main__":
    app.run()
```

**Comment by akshayka** [comment]
Thanks for reporting; we'll look into it

**Comment by mgunyho** [comment]
Hi, thanks for fixing the asyncio error! However, looks like the infinitely growing plot issue is still not resolved in 0.5.1. The minimal code to reproduce it is a single cell

```python
import marimo as mo
import matplotlib.pyplot as plt

mo.mpl.interactive(plt.figure().gca())
```


Here's the output of `marimo env`, after creating a fresh virtual environment and running `pip install marimo matplotlib tornado`:

```json
{
  "marimo": "0.5.1",
  "OS": "Linux",
  "OS Version": "5.15.0-105-generic",
  "Processor": "x86_64",
  "Python Version": "3.8.10",
  "Binaries": {
    "Browser": "124.0.6367.201",
    "Node": "--"
  },
  "Requirements": {
    "click": "8.1.7",
    "importlib-resources": "6.4.0",
    "jedi": "0.19.1",
    "markdown": "3.6",
    "pymdown-extensions": "10.8.1",
    "pygments": "2.18.0",
    "tomlkit": "0.12.5",
    "uvicorn": "0.29.0",
    "starlette": "0.37.2",
    "websocket": "missing",
    "typing-extensions": "4.11.0",
    "black": "24.4.2"
  }
}
```

I found some possibly related issues: #1072, #1088, but the issue is still not resolved for me.
